### PR TITLE
Added hashicorp Library

### PR DESCRIPTION
**Pull request by rain-on**
This includes the required gradle files to build and distribute the jar files.

### CODE REVIEW

File: keystorage/hashicorp/src/main/java/tech/pegasys/signing/hashicorp/HashicorpConnection.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+  private final HttpClient httpClient;
+  private final long requestTimeoutMs;
+
+  public HashicorpConnection(final HttpClient httpClient, final long requestTimeoutMs) {
+    this.httpClient = httpClient;
+    this.requestTimeoutMs = requestTimeoutMs;
+  }
+
+  public String fetchKey(final KeyDefinition key) {
```
Comment (jframe) [inline keystorage/hashicorp/src/main/java/tech/pegasys/signing/hashicorp/HashicorpConnection.java]: This name is bit confusing to me, as it's not fetching the key is fetching the value for the value? Perhaps just fetch would be sufficient

File: keystorage/hashicorp/src/main/java/tech/pegasys/signing/hashicorp/HashicorpConnection.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+import javax.net.ssl.SSLException;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientResponse;
+
+public class HashicorpConnection {
```
Comment (jframe) [inline keystorage/hashicorp/src/main/java/tech/pegasys/signing/hashicorp/HashicorpConnection.java]: Name of this class doesn't quite seem right to me. The methods are doing anything with the connection, there is no connect or disconnect. Thinking it's just the HashicorpVault?

File: keystorage/hashicorp/src/main/java/tech/pegasys/signing/hashicorp/HashicorpConnectionFactory.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            .setDefaultHost(connectionParameters.getServerHost())
+            .setDefaultPort(serverPort);
+
+    final HttpClient httpClient;
+    try {
+      if (connectionParameters.getTlsOptions().isPresent()) {
+        LOG.debug("Connection to hashicorp vault using TLS.");
+        setTlsOptions(httpClientOptions, connectionParameters.getTlsOptions().get());
+      }
+      httpClient = vertx.createHttpClient(httpClientOptions);
```
Comment (jframe) [inline keystorage/hashicorp/src/main/java/tech/pegasys/signing/hashicorp/HashicorpConnectionFactory.java]: could just do the return HashicorpConnection here and avoid the need to declare the httpClient before the try

File: keystorage/hashicorp/src/testFixtures/java/tech/pegasys/signing/hashicorp/dsl/hashicorp/HashicorpVaultCommands.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    return new String[] {
+      "vault", "kv", "put", "-address=" + vaultUrl, path, paramString,
+    };
+  }
+
+  public String[] vaultLoginCommand(final String rootToken) {
+    return new String[] {"vault", "login", "-address=" + vaultUrl, rootToken};
+  }
+
+  public List<String> constructVaultEnvVars() {
```
Comment (jframe) [inline keystorage/hashicorp/src/testFixtures/java/tech/pegasys/signing/hashicorp/dsl/hashicorp/HashicorpVaultCommands.java]: delete commented out code

File: keystorage/hashicorp/build.gradle [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  archiveBaseName = "hashicorp-vault"
+  manifest {
+    attributes(
+      'Specification-Title': archiveBaseName.get(),
+      'Specification-Version': rootProject.version,
+      'Implementation-Title': archiveBaseName.get(),
+      'Implementation-Version': calculateVersion()
+      )
+  }
+  from {
```
Comment (jframe) [inline keystorage/hashicorp/build.gradle]: This shouldn't be done for the signer library

File: keystorage/hashicorp/build.gradle [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+ * The source code is provided to licensees of PegaSys Plus for their convenience and internal
+ * business use. These files may not be copied, translated, and/or distributed without the express
+ * written permission of an authorized signatory for ConsenSys AG.
+ */
+
+apply plugin: 'java-library'
+apply plugin: 'java-test-fixtures'
+
+jar {
+  archiveBaseName = "hashicorp-vault"
```
Comment (jframe) [inline keystorage/hashicorp/build.gradle]: Think this should match the module name, so keystore-hashicorp instead?

File: keystorage/hashicorp/build.gradle [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  implementation 'org.apache.logging.log4j:log4j-api'
+
+  implementation 'org.apache.tuweni:tuweni-toml'
+  implementation 'org.apache.tuweni:tuweni-net'
+  implementation 'org.bouncycastle:bcpkix-jdk15on'
+
+  runtimeOnly 'org.apache.logging.log4j:log4j-core'
+  runtimeOnly 'org.apache.logging.log4j:log4j-slf4j-impl'
+
+  annotationProcessor 'com.google.auto.service:auto-service'
```
Comment (jframe) [inline keystorage/hashicorp/build.gradle]: This can be removed too as aren't using java modules

File: keystorage/hashicorp/build.gradle [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  }
+  from {
+    configurations.compile.collect {
+      it.isDirectory() ? it : zipTree(it)
+    }
+  }
+}
+
+dependencies {
+  implementation 'com.google.auto.service:auto-service'
```
Comment (jframe) [inline keystorage/hashicorp/build.gradle]: This can be removed too as aren't using java modules

**Comment by rain-on** [comment]
Splitting this into smaller PRs

**Pull request closed without merging**